src/bin/sol-fbp-generator/main.c

```diff
@@ -634,7 +634,15 @@ generate_node_type_assignments(const struct fbp_data *data)
 
     SOL_VECTOR_FOREACH_IDX (&data->graph.nodes, n, i) {
         struct type_description *desc = n->user_data;
-        out("    nodes[%d].type = %s;\n", i, desc->symbol);
+        struct sol_str_slice module;
+        char *sep;
+
+        module = sol_str_slice_from_str(desc->name);
+        sep = strstr(module.data, "/");
+        if (sep)
+            module.len = sep - module.data;
+        out("    nodes[%d].type = *(const struct sol_flow_node_type **)sol_symbol_get(\"flow\", \"%.*s\", %s);\n", i,
```

Comment:
I don't get why we need to derreference `*(x **)`. Also, shouldn't it check `NULL` in the case of symbol is missing (no module, etc)? It would be the only reason to keep the derreference.

If we're going to fail badly -- such as abort, then remove the derreference and provide a nice error message inside `sol_symbol_get()`